src/git-command-manager.ts

```diff
@@ -357,6 +358,12 @@ class GitCommandManager {
     await this.execGit(args)
   }
 
+  async submoduleStatus(): Promise<boolean> {
+    const output = await this.execGit(['submodule', 'status'], true)
+    core.debug(output.stdout)
+    return output.exitCode === 0
+  }
+
```

Comment:
for more info on the `git submodule` command, see:  https://git-scm.com/docs/git-submodule#Documentation/git-submodule.txt-status--cached--recursive--ltpathgt82308203

Reply:
@jww3  This can be recreated by cloning https://github.com/test-SKi/test-bad-module-clone & using checkout action with any self-hosted runner to see the failure.

### Steps to recreate one scenario:
```
git checkout -b new
git clone git@github.com:test-SKi/submodule-repo.git
echo "recreate bad submodule issue" >>  README.md
git add .
git commit
git push --set-upstream origin new
```